### PR TITLE
python3Packages.ansible-doctor: init at 1.2.1

### PR DESCRIPTION
**Pull request by tboerger**
###### Description of changes

Init new python module for generating documentation of Ansible roles. Depends on https://github.com/NixOS/nixpkgs/pull/165283 and https://github.com/NixOS/nixpkgs/pull/165286

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/ansible-doctor/default.nix [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    python-json-logger
+    pathspec
+    environs
+    jsonschema
+    appdirs
+    ruamel-yaml
+    anyconfig
+    nested-lookup
+  ];
+
```
Comment (jonringer) [inline pkgs/development/python-modules/ansible-doctor/default.nix:56]: If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.

Comment (tboerger) [reply pkgs/development/python-modules/ansible-doctor/default.nix:56]: Upstream is executing poetry run ansible-doctor --help as a test. Does it make sense to do that here as well?

File: pkgs/development/python-modules/ansible-doctor/default.nix [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  doCheck = false;
+  pythonImportsCheck = [ "ansibledoctor" ];
+
+  meta = with lib; {
+    description = "Annotation based documentation for your Ansible roles";
+    homepage = "https://github.com/thegeeklab/ansible-doctor";
+    license = licenses.lgpl3Only;
+    maintainers = with maintainers; [
+      tboerger
+    ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ansible-doctor/default.nix:66]:
```suggestion
    maintainers = with maintainers; [ tboerger ];
```

File: pkgs/development/python-modules/ansible-doctor/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/ansible-doctor/default.nix:56]: Not really...

```suggestion

  # no tests
```

**Comment by tboerger** [comment]
@SuperSandro2000 I have applied your requested changes.

File: pkgs/development/python-modules/ansible-doctor/default.nix [hunk 3]
```diff
@@ -0,0 +1,67 @@
+{ buildPythonPackage
+, fetchFromGitHub
+, lib
+
+# pythonPackages
+, appdirs
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ansible-doctor/default.nix:6]:
```suggestion
, anyconfig
, appdirs
```

File: pkgs/development/python-modules/ansible-doctor/default.nix [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+, fetchFromGitHub
+, lib
+
+# pythonPackages
+, anyconfig
+, appdirs
+, colorama
+, environs
+, jinja2
+, jsonschema
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ansible-doctor/default.nix:11]:
```suggestion
, jsonschema
, nested-lookup
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 165289` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>python310Packages.ansible-doctor</li>
<li>python39Packages.ansible-doctor</li>
</ul>
</details>

**Pull request merged**